### ISSUE TITLE
Incompatibility with package db-dtypes results in casts of object to string types

### ISSUE THREAD

**Issue by HHarald99** [dask/dask#10494]
**Describe the issue**:

When the package [db-dtypes](https://pypi.org/project/db-dtypes/) is installed dask will convert object-type columns (at least those containing arrays) to strings when converting a pandas dataframe to a dask dataframe. 

This is unfortunate as db-dtypes is important when using BigQuery with Python. Arrays within cells ("repeated fields" in Bigquery) are also quite common.

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
import numpy as np
import dask.dataframe as dd
import dask.bag as db
import dask

if __name__ == "__main__":
    df = pd.DataFrame({"a": [np.array([5, 6, 7, 8])]})
    print(type(df.iloc[0]["a"]))
    df_dask = dd.from_pandas(df, npartitions=1)
    print(type(df_dask.compute().iloc[0]["a"]))
```
This usually yields

```
<class 'numpy.ndarray'>
<class 'numpy.ndarray'>
```

As soon as db-dtypes is installed, the result is

```
<class 'numpy.ndarray'>
<class 'str'>
```

**Anything else we need to know?**:
Minimal requirements.txt:
```
dask==2023.9.0
db-dtypes==1.1.1
```

**Environment**:

- Dask version:
2023.9.0
- Python version:
3.10
- Operating System: 
  OS x
- Install method (conda, pip, source): 
  pip

**Comment by shezadkhan137** [comment]
I just ran into this issue when using `map_partitions`. Any dtype=object column in a pd.DataFrame returned from `map_partitions` are casted to a string, making it impossible have a DataFrame column that contains lists, np.arrays etc

It doesn't seem to be anything to do with [db-dtypes](https://pypi.org/project/db-dtypes/). I can reproduce by installing **any** dask version >= [2023.7.1](https://github.com/dask/dask/compare/2023.7.0...2023.7.1), along with pandas >= 2.0. 

It looks like its related to automatically converting the dtype=object to a pyarrow string type.  Specifically the [is_object_string_dtype](https://github.com/dask/dask/compare/2023.7.0...2023.7.1#diff-b7e83c2b55f86ffb06a4d3d088527f622b2de52899f30ebdec134789150b26acR29-R36) returns true for any object dtype, which then later on the [code tries to cast to a string](https://github.com/dask/dask/compare/2023.7.0...2023.7.1#diff-b7e83c2b55f86ffb06a4d3d088527f622b2de52899f30ebdec134789150b26acR76).

**Comment by phofl** [comment]
Yep we started converting object to string by default in 2023.07.1, you can turn this off through

```
dask.config.set({"dataframe.convert-string": False})
```
